ledger/complete/wal/wal_test.go

```diff
@@ -115,3 +144,26 @@ func Test_NoGapBetweenSegmentsAndLastCheckpoint(t *testing.T) {
 		require.Error(t, err)
 	})
 }
+
+func Test_LatestPossibleCheckpoints(t *testing.T) {
+
+	require.Equal(t, []int(nil), getPossibleCheckpoints([]int{1, 2, 5}, 0, 0))
+
+	require.Equal(t, []int{1}, getPossibleCheckpoints([]int{1, 2, 5}, 0, 1))
+	require.Equal(t, []int{1, 2}, getPossibleCheckpoints([]int{1, 2, 5}, 0, 2))
+	require.Equal(t, []int{1, 2}, getPossibleCheckpoints([]int{1, 2, 5}, 0, 3))
+	require.Equal(t, []int{1, 2}, getPossibleCheckpoints([]int{1, 2, 5}, 0, 4))
+
+	require.Equal(t, []int{1, 2, 5}, getPossibleCheckpoints([]int{1, 2, 5}, 0, 5))
+	require.Equal(t, []int{1, 2, 5}, getPossibleCheckpoints([]int{1, 2, 5}, 0, 6))
+	require.Equal(t, []int{1, 2, 5}, getPossibleCheckpoints([]int{1, 2, 5}, 1, 6))
+
+	require.Equal(t, []int{2, 5}, getPossibleCheckpoints([]int{1, 2, 5}, 2, 6))
+	require.Equal(t, []int{2}, getPossibleCheckpoints([]int{1, 2, 5}, 2, 3))
+	require.Equal(t, []int{2}, getPossibleCheckpoints([]int{1, 2, 5}, 2, 4))
+	require.Equal(t, []int{2, 5}, getPossibleCheckpoints([]int{1, 2, 5}, 2, 5))
+
+	require.Equal(t, []int{5}, getPossibleCheckpoints([]int{1, 2, 5}, 3, 5))
+	require.Equal(t, []int{5}, getPossibleCheckpoints([]int{1, 2, 5}, 3, 6))
+
```

Comment:
are:
```
	require.Equal(t, []int{5}, getPossibleCheckpoints([]int{1, 2, 5}, 5, 5))
	require.Equal(t, []int(nil), getPossibleCheckpoints([]int{1, 2, 5}, 6, 6))
```
useful test cases at all?

Reply:
I think so, as they tests for exact matching of bounds in the first case, and out of bounds total in another. Can you elaborate

Reply:
I think @Kay-Zee is suggesting to _add_ two more test cases, instead of removing some, which makes sense to me.

Reply:
ah, right! I added them, thanks!